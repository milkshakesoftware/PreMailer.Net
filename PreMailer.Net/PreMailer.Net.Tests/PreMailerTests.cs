using System;
using ApprovalTests.Reporters;
using System.IO;
using NUnit.Framework;

namespace PreMailer.Net.Tests
{
	[TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class PreMailerTests
	{
		private PreMailer _sut;

		[SetUp]
		public void TestInitialize()
		{
			_sut = new PreMailer();
		}

		[Test]
		public void MoveCssInline_RespectExistingStyleElement()
		{
			string input = "<html><head><style type=\"text/css\">.test { height: 100px; }</style></head><body><div class=\"test\" style=\"width: 100px;\">test</div></body></html>";

			string premailedOutput = _sut.MoveCssInline(input, false);

            Console.WriteLine(premailedOutput);
            Assert.IsTrue(premailedOutput.Contains("<div class=\"test\" style=\"width: 100px; height: 100px;"));
		}

		[Test]
		public void MoveCssInline_InlineStyleElementTakesPrecedence()
		{
			string input = "<html><head><style type=\"text/css\">.test { width: 150px; }</style></head><body><div class=\"test\" style=\"width: 100px;\">test</div></body></html>";

			string premailedOutput = _sut.MoveCssInline(input, false);

            Console.WriteLine(premailedOutput);
			Assert.IsTrue(premailedOutput.Contains("<div class=\"test\" style=\"width: 100px"));
		}

		[Test]
		public void ManualIntegrationTest()
		{
			string htmlSource = File.ReadAllText("testmail.html");

			string premailedOutput = _sut.MoveCssInline(htmlSource, false);
            Console.WriteLine(premailedOutput);

            ApprovalTests.Approvals.Verify(premailedOutput);
        }
	}
}