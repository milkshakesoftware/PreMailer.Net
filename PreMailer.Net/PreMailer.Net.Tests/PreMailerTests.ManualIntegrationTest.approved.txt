<!DOCTYPE html "-//w3c//dtd xhtml 1.0 transitional//en" "http://www.w3.org/tr/xhtml1/dtd/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	<style type="text/css">
		body
		{
			background-color: #f1f1f1;
			font-family: Arial, Helvetica;
		}
		.backgroundTable
		{
			background-color: #f1f1f1;
		}
		.wrapper
		{
			width: 600px;
		}
		.onecolumn
		{
			width: 600px;
		}
		.dummy-row
		{
			display: none;
		}
		.section-table-header img
		{
			width: 600px;
			height: 80px;
		}
		.footer
		{
			font-size: 12px;
			color: #454545;
		}
	</style>
</head>
<body style="background-color: #f1f1f1; font-family: arial, helvetica;" leftmargin="0" marginwidth="0" topmargin="0" marginheight="0" offset="0">
	<table class="backgroundTable" style="background-color: #f1f1f1" width="100%" cellpadding="0" cellspacing="0">
		<tbody><tr>
			<td valign="top" align="center">
				<table class="wrapper onecolumn" style="width: 600px" width="600">
					<tbody><tr class="header">
						<td>
							<table class="section-table-header">
								<tbody><tr class="dummy-row" style="display: none">
									<td class="grid-dummy-cell-1">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-2">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-3">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-4">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-5">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-6">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-7">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-8">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-9">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-10">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-11">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-12">
										&nbsp;
									</td>
								</tr>
								<tr class="section-row-header-1">
									<td class="section-column-header-1" colspan="12">
										<img style="height: 80px; width: 600px;" alt="milkshake-commerce-email-header" src="http://apps.milkshakecommerce.com/email/milkshake-commerce-email-header.png" />
									</td>
								</tr>
							</tbody></table>
						</td>
					</tr>
					<tr class="layout-row">
						<td>
							<table class="content-section">
								<tbody><tr class="dummy-row" style="display: none">
									<td class="grid-dummy-cell-1">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-2">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-3">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-4">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-5">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-6">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-7">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-8">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-9">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-10">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-11">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-12">
										&nbsp;
									</td>
								</tr>
								<tr class="precontent-section">
								</tr>
								<tr class="content-row-1">
									<td class="content-column-1-1" colspan="12">
										<h1>
											Test that shit now</h1>
										<p>
											aa asd as d</p>
										<p>
											as d</p>
										<p>
											as asdas asdas adsd as</p>
									</td>
								</tr>
								<tr class="postcontent-section">
								</tr>
							</tbody></table>
						</td>
					</tr>
					<tr class="footer" style="color: #454545; font-size: 12px;">
						<td>
							<table class="section-table-footer">
								<tbody><tr class="dummy-row" style="display: none">
									<td class="grid-dummy-cell-1">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-2">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-3">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-4">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-5">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-6">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-7">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-8">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-9">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-10">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-11">
										&nbsp;
									</td>
									<td class="grid-dummy-cell-12">
										&nbsp;
									</td>
								</tr>
								<tr class="section-row-footer-1">
									<td class="section-column-footer-1" colspan="12">
										<p>
											Denne e-mail er sendt fra Milkshake Commerce. Hvis du mener at have modtaget e-mail'en
											ved en fejl, bedes du venligst videresende den til: support@milkshakecommerce.com</p>
									</td>
								</tr>
								<tr class="section-row-footer-2">
									<td class="section-column-footer-1" colspan="12">
										<p>
											<a href="http://www.milkshakecommerce.com">Milkshake Commerce</a> er en l&#230;kker e-handelsplatform,
											som ogs&#229; indeholder v&#230;rkt&#248;jer til markedsf&#248;ring s&#229; du kan oprette og markedsf&#248;re
											en successful forretning p&#229; nettet.</p>
									</td>
								</tr>
							</tbody></table>
						</td>
					</tr>
				</tbody></table>
			</td>
		</tr>
	</tbody></table>
</body></html>